src/validatedODEs.jl

```diff
@@ -0,0 +1,332 @@
+        x::Vector{Taylor1{TaylorModelN{N,R,T}}},
+        dx::Vector{Taylor1{TaylorModelN{N,R,T}}},
+        xaux::Vector{Taylor1{TaylorModelN{N,R,T}}},
+        t0::T, t1::T,
+        orderT::Int, abstol::T,
+        parse_eqs::Bool=true) where {N,R,T<:Real}
+
+    @assert t1 > t0
+
+    # Compute the Taylor coefficients (non-validated integration)
+    TaylorIntegration.__jetcoeffs!(Val(parse_eqs), f!, t, x, dx, xaux)
+
+    # Compute the step-size of the integration using `abstol`
+    δt = TaylorIntegration.stepsize(x, abstol)
+    δt = min(δt, t1-t0)
+
+    return δt
+end
+
+
+function validated_integ(f!, q0::IntervalBox{N,T}, δq0::IntervalBox{N,T},
+        t0::T, tmax::T, orderQ::Int, orderT::Int, abstol::T;
+        maxsteps::Int=500, parse_eqs::Bool=true, sym_norm::Bool=true,
+        check_property::Function=x->true) where {N, T<:Real}
+
+    # Set proper parameters for jet transport
+    @assert N == get_numvars()
+    dof = N
+    if get_order() != 2*orderQ
+        set_variables("δ", numvars=dof, order=2*orderQ)
+    end
+
+    # Some variables
+    R   = Interval{T}
+    ti0 = Interval(t0)
+    t   = ti0 + Taylor1(orderT)
+    δq_norm = sym_norm ? IntervalBox(-1..1, Val(N)) : IntervalBox(0..1, Val(N))
+
+    # Allocation of vectors
+    # Output
+    tv    = Array{T}(undef, maxsteps+1)
+    xv    = Array{IntervalBox{N,T}}(undef, maxsteps+1)
+    xTMNv = Array{TaylorModelN{N,R,T}}(undef, dof)
+    # Internals
+    x     = Array{Taylor1{TaylorModelN{N,R,T}}}(undef, dof)
+    dx    = Array{Taylor1{TaylorModelN{N,R,T}}}(undef, dof)
+    xaux  = Array{Taylor1{TaylorModelN{N,R,T}}}(undef, dof)
+    xTMN  = Array{TaylorModelN{N,R,T}}(undef, dof)
+
+    # Set initial conditions
+    zI = zero(R)
+    Δ = zero.(q0)
+    @inbounds for i in eachindex(x)
+        qaux = normalize_taylor(q0[i] + TaylorN(i, order=orderQ), δq0, sym_norm)
+        x[i] = Taylor1( TaylorModelN(qaux, zI, q0, q0 + δq_norm), orderT )
+        dx[i] = x[i]
+        xTMN[i] = x[i][0]
+    end
+
+    # Output vectors
+    @inbounds tv[1] = t0
+    @inbounds xv[1] = IntervalBox( evaluate(xTMN, δq_norm) )
+    @inbounds xTMNv .= xTMN
+
+    # Determine if specialized jetcoeffs! method exists (built by @taylorize)
+    parse_eqs = parse_eqs && (length(methods(TaylorIntegration.jetcoeffs!)) > 2)
+    if parse_eqs
+        try
+            TaylorIntegration.jetcoeffs!(Val(f!), t, x, dx)
+        catch
+            parse_eqs = false
+        end
+    end
+
+    # Integration
+    nsteps = 1
+    while t0 < tmax
+        # One step integration (non-validated)
+        δt = TaylorIntegration.taylorstep!(f!, t, x, dx, xaux,
+            t0, tmax, orderT, abstol, parse_eqs)
+
+        # This is to compute dx[:][orderT] (now zero), needed for the remainder
+        f!(t, x, dx)
+
+        # Perform 25 checks of `check_property`; if the property is not satisfied
+        # throw an error
+        nsteps += 1
+        issatisfied = false
+        for nchecks = 1:25
```

Comment:
Is there a fundamental reason for 25? Maybe make this a parameter.

Reply:
Thanks for taking a look on this, @schillic !

No, there is actually no reason to have it 25. Yet, the heuristic argument is the following: 1/2^25 ≈ 3e-8, which is the smallest factor by which the original δt would be reduced, which seems reasonable. (To have it 30 means a factor close to 1.0e-9.)

 For the time being, this is purely experimental: we have to find out a good way to reduce time when it is needed. Experimenting with Laub-Loomis system has showed that it is not yet enough.